es6-lib/decoders/index.js

```diff
@@ -3,13 +3,23 @@ import GeoJSON from './geojson';
 import KML from './kml';
 import KMZ from './kmz';
 import Shapefile from './shapefile';
+import path from 'path';
 
-function getDecoder(req, disk) {
+function getDecoderForContentType(req, disk) {
   var ctype = req.headers['content-type'];
   var decoder = _.find([GeoJSON, KML, KMZ, Shapefile], (k) => _.contains(k.canDecode(), ctype));
   if(!decoder) return [new Error(`No decoder found for ${ctype}`), false];
   req.log.info(`Selected decoder ${decoder.name} to parse request`);
   return [false, new decoder(disk)];
 }
 
-export default getDecoder;
\ No newline at end of file
+function getDecoderForExtension(filename, disk) {
+  const extension = path.extname(filename);
```

Comment:
Good call, These are always lowercase from core, but i've added a `toLowerCase` so we don't break it if we change core